tests/Unit/Helpers/DataStructures/Tensor/Expressions/EvaluateRank1TestHelpers.hpp

```diff
@@ -33,14 +34,18 @@ void test_evaluate_rank_1_impl() noexcept {
   // L_a = R_a
   // Use explicit type (vs auto) so the compiler checks return type of
   // `evaluate`
-  const Tensor<DataType, Symmetry<1>, TensorIndexTypeList> L_a =
+  const Tensor<DataType, Symmetry<1>, TensorIndexTypeList> L_a_returned =
       ::TensorExpressions::evaluate<TensorIndex>(R_a(TensorIndex));
+  Tensor<DataType, Symmetry<1>, TensorIndexTypeList> L_a_filled{};
```

Comment:
[choosing a test at random]: could you add a test where the passed in `Tensor` has non-owning `DataVector`s? The best example would be:
```cpp
Variables<tmpl::list<Tags::Tempa<...>>> vars{num_grid_points};
::TensorExpressions::evaluate<TensorIndex>(make_not_null(&get<Tags::Tempa<...>>(vars)),
                                             R_a(TensorIndex));
```
`Tags::Tempa` is in `DataStructures/Tags/TempTensor.hpp`. This is to make sure that a very common use case that helps eliminate memory allocations works correctly. The code _looks_ like it'll work, but I can't compile and run the test in my head :P

Reply:
Yep! Should this be for only the `DataVector` test cases and not `double`?

Reply:
Gave this a shot in the 3rd fixup commit